weblab_cg/_generator_chaste_cg.py

```diff
@@ -0,0 +1,627 @@
+            A :class:`cellmlmanip.Model` object.
+        ``output_path``
+            The path to store the generated model code at.
+            (Just the path, excluding the file name as file name will be determined by the model_name)
+        ``model_name_for_filename``
+            The model name as you want it to appear in generated cpp and hpp files.
+        """
+        self._logger = logging.getLogger(__name__)
+        self._logger.setLevel(logging.INFO)
+
+        # Store parameters for future reference
+        self._model = model
+
+        self._model_name_for_filename = model_name_for_filename
+        self._class_name = class_name
+        self.generated_hpp = ''
+        self.generated_cpp = ''
+
+        self._add_units()
+
+        self._time_variable = self._model.get_free_variable_symbol()
+        self._membrane_voltage_var = self._get_membrane_voltage_var()
+        self._cytosolic_calcium_concentration_var = self._get_cytosolic_calcium_concentration_var()
+        self._use_get_intracellular_calcium_concentration = self._get_use_get_intracellular_calcium_concentration()
+        self._state_vars = self._get_state_variables()
+        self._membrane_stimulus_current = self._get_membrane_stimulus_current()
+        self._membrane_stimulus_current_units = self._get_membrane_stimulus_current_units()
+        self._membrane_capacitance, self._membrane_capacitance_factor = self._get_membrane_capacitance()
+        self._default_stimulus = self._get_stimulus_currents()
+        self._desired_ionic_current_units, self._ionic_current_factor = self._get_stimulus_currents_desired_units()
+        self._ionic_derivs = self._get_ionic_derivs()
+        self._equations_for_ionic_vars = self._get_equations_for_ionic_vars()
+        self._extended_equations_for_ionic_vars = self._get_extended_equations_for_ionic_vars()
+
+        self._y_derivatives = self._get_y_derivatives()
+        self._y_derivatives_voltage = self._get_y_derivatives_voltage()
+        self._y_derivatives_excl_voltage = self._get_y_derivatives_excl_voltage()
+        self._derivative_eqs_voltage = self._get_derivative_eqs_voltage()
+        self._derivative_eqs_exlc_voltage = self._get_derivative_eqs_exlc_voltage()
+        self._derivative_equations = self._get_derivative_equations()
+
+        self._add_printers()
+        self._formatted_state_vars = self._format_state_variables()
+        self._formatted_default_stimulus = self._format_stimulus_currents()
+        self._formatted_equations_for_ionic_vars = self._format_equations_for_ionic_vars()
+        self._formatted_extended_equations_for_ionic_vars = self._format_extended_equations_for_ionic_vars()
+        self._formatted_y_derivatives = self._format_y_derivatives()
+        self._formatted_derivative_eqs = self._format_derivative_equations()
+        self._free_variable, self._ode_system_information, self._named_attributes = self._format_system_info()
+
+    def generate_chaste_code(self):
```

Comment:
If I can't call it, then why is this here?

Reply:
To make sure you get a useful and descriptive error message  if you accidentally call it on the wrong type of model. Also I like to be able to see from a base class if there is a method all subclasses will have

Reply:
Woops, sorry didn't notice this was a base class

Reply:
lol no worries